src/SheetsRegistry.js

```diff
@@ -15,7 +15,21 @@ export default class SheetsRegistry {
    * @api public
    */
   add(sheet) {
-    this.registry.push(sheet)
+    const {registry} = this
+
+    if (registry.length && sheet.options.index) {
+      for (let i = 0; i < registry.length; i++) {
+        const registration = registry[i]
```

Comment:
oh I see, it conflicts with outer sheet const, maybe call the first sheet "newSheet" ?